### DISCUSSION TITLE
Does KafkaUI support OAuth token refresh?

### DISCUSSION THREAD

**Discussion by krumft** [provectus/kafka-ui#3773] [General]
Hello KafkaUI Team,

Thank you for the great tool!

We are deploying it in a setup that uses our corporate Okta tenant for authentication and authorisation. The type of integration we have configured on Okta level is an OIDC web app. Our tenant supports the `refresh_token` grant type, looking at the response from its `.well-known/openid-configuration` endpoint.

We have noticed that every couple of hours the application session expires, and the user is redirected to a brand new SSO login journey. Therefore I am opening this discussion to ask following questions:

- In an OAuth setting how long does the user session last, and is that duration configurable?
- (kind of similar question) How long does the OAuth access token last? Usually this is specified as a metadata of the token, and is controlled by the party that issues the token, so perhaps KafkaUI just respects that metadata?
- Does KafkaUI support OAuth token refresh? Or every time the user session expires it requests a brand new access token?

Kind Regards,
Krum.

**Comment by Haarolean** [comment]
Not solvable out of the box as far as I can see. I'll convert this to an issue.

**Comment by Haarolean** [comment]
#3799
